File: university_projects/car_rental/android_version/java/com/example/wiktorpieklik/car_rental/Model/Order.java
package com.example.wiktorpieklik.car_rental.Model;

public class Order
{
    private int orderId;
    private int carID;
    private int clientID;
    private String dateOfPurchase;
    private String dateOfReturn;
    private float price;
    private float discount;
    private int discountGranted;

    public Order(int id, String dateOfPurchase, String dateOfReturn, float price, float discount, int discountGranted, int carID, int clientID)
    {
        orderId=id;
        this.dateOfPurchase=dateOfPurchase;
        this.dateOfReturn=dateOfReturn;
        this.price=price;
        this.discount=discount;
        this.discountGranted=discountGranted;
        this.carID=carID;
        this.clientID=clientID;
    }

    public int getOrderId()
    {
        return this.orderId;
    }

    public String getOrderDateOfPurchase()
    {
        return this.dateOfPurchase;
    }

    public String getOrderDateOfReturn()
    {
        return this.dateOfReturn;
    }

    public float getOrderPrice()
    {
        return this.price;
    }

    public float getOrderDiscount()
    {
        return this.discount;
    }

    public int getDiscountGrantedState()
    {
        return this.discountGranted;
    }

    public int getOrderClientID()
    {
        return this.clientID;
    }

    public int getOrderCarID()
    {
        return this.carID;
    }
}

File: university_projects/car_rental/cpp_version/Order.cpp
#include "Order.h"

Order::Order(int id, String^ dateOfPurchase, String^ dateOfReturn, float price, float discount, Boolean discountGranted, int carID, int clientID) :orderId(id), dateOfPurchase(dateOfPurchase), dateOfReturn(dateOfReturn), price(price), discount(discount), discountGranted(discountGranted), carID(carID), clientID(clientID)
{
}


int Order::getOrderId()
{
	return this->orderId;
}


String^ Order::getOrderDateOfPurchase()
{
	return this->dateOfPurchase;
}


String^ Order::getOrderDateOfReturn()
{
	return this->dateOfReturn;
}


float Order::getOrderPrice()
{
	return this->price;
}


float Order::getOrderDiscount()
{
	return this->discount;
}


Boolean Order::getDiscountGrantedState()
{
	return this->discountGranted;
}

int Order::getOrderClientID()
{
	return this->clientID;
}

int Order::getOrderCarID()
{
	return this->carID;
}